File: services/utils-service/src/main/java/com/sgpttt/UtilsService/model/ProtocolState.java
package com.sgpttt.UtilsService.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class ProtocolState {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long state_id;

    private String name;

    public ProtocolState() {
    }

    public ProtocolState(String name) {
        this.name = name;
    }

    public Long getState_id() {
        return state_id;
    }
    public void setState_id(Long state_id) {
        this.state_id = state_id;
    }

    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
}

File: services/protocols-service/src/main/kotlin/com/sgptt/protocolsservice/repository/entity/ProtocolState.kt
package com.sgptt.protocolsservice.repository.entity

import com.sgptt.protocolsservice.repository.entity.model.ProtocolState
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "protocolstate")
data class ProtocolState(
	
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	@Column(name = "state_id")
	val id: Long,
	
	@Enumerated(EnumType.STRING)
	@Column(name = "name", length = 50)
	val state: ProtocolState
)
